src/phases/move-phase.ts

```diff
@@ -225,34 +225,37 @@ export class MovePhase extends BattlePhase {
    * Handles {@link StatusEffect.SLEEP Sleep}/{@link StatusEffect.PARALYSIS Paralysis}/{@link StatusEffect.FREEZE Freeze} rolls and side effects.
    */
   protected resolvePreMoveStatusEffects(): void {
-    if (!this.followUp && this.pokemon.status && !this.pokemon.status.isPostTurn()) {
-      this.pokemon.status.incrementTurn();
+    if (
+      !this.followUp
+      && this.pokemon.hasStatusEffect([StatusEffect.SLEEP, StatusEffect.PARALYSIS, StatusEffect.FREEZE], false, true)
+    ) {
+      this.pokemon.status!.incrementTurn();
       let activated = false;
       let healed = false;
 
-      switch (this.pokemon.status.effect) {
+      switch (this.pokemon.getStatusEffect(true)) {
```

Comment:
```suggestion
      const statusEffect = this.pokemon.getStatusEffect(true);
      switch (statusEffect) {
```
you may want to double check that it's not breaking anything, but it's probably easier to use a const here to avoid potential mistakes like forgetting the `true` in the call to getStatusEffect